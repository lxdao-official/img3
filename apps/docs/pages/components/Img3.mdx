---
title: Img3
description: <Img3 /> is a fundamental components for Web3 Apps. It extends HTML < img /> with Web3 decentralization storage, like IPFS. With <Img3 /> you can put ipfs:// in the src and render the image from IPFS with the fastest gateway. Will support ar soon.
---

# Img3

`<Img3 />` is a fundamental components for Web3 Apps. It extends HTML `<img />` with Web3 decentralization storage, like
IPFS. With `<Img3 />` you can put ipfs:// in the src and render the image from IPFS with the fastest gateway. Will support
ar soon.

**Features**

- Supports the `ipfs://` protocol.
- Automatically uses the gateway with the highest speed, And reuse it the next request.

<Anchor id="defaultGateways" />

```jsx
const defaultGateways = [
  'https://nftstorage.link/ipfs/',
  'https://ipfs-gateway.cloud/ipfs/',
  'https://gateway.pinata.cloud/ipfs/',
  'https://4everland.io/ipfs/',
];
```

## Usage

```js copy
import { Img3 } from '@lxdao/img3';
```

## Props

| Prop     | Type            | Description                              | Default                             |
| -------- | --------------- | ---------------------------------------- | ----------------------------------- |
| src      | `string`        | image source, support `ipfs://` protocol |                                     |
| alt      | `string`        | image alt text                           |                                     |
| gateways | `string[]`      | Web3 decentralization storage gateways   | [defaultGateways](#defaultGateways) |
| timeout  | `number`        | timeout of gateway request               | `2000`                              |
| icon     | [`Icon`](#icon) | style of loading or error icons          | `{size: 30, color: '#c0c0c0'}`      |

### Img3Provider

`@lxdao/img3` provides an `Img3Provider` provider component that should be used to wrap around your components.

`Img3Provider`'s responsibility is to pass the default gateway value to all of the `<Img3 />` components in your application.
Usually, your application only needs one provider. You should place it as high as possible in your React tree.

| Prop           | Type     | Description                           | Default |
| -------------- | -------- | ------------------------------------- | ------- |
| defaultGateway | `string` | Web3 decentralization storage gateway |         |

```ts
import React from 'react';
import { Img3Provider } from '@lxdao/img3';

export default function Demo() {
  return (
    <Img3Provider defaultGateway="https://nftstorage.link/ipfs/">
      <YourApp />
    </Img3Provider>
  );
}
```

## Types reference

### Icon

```ts
type Icon = {
  /** icon size. */
  size?: number;
  /** icon color. */
  color?: string;
  /** error icon size. */
  errorSize?: number;
  /** error icon color. */
  errorColor?: string;
};
```

## Examples

import img3Examples from '@/examples/img3/';

### Basic

Like `<img />` tag, set the `src` to render the image.

<Playground title={'Basic - @lxdao/img3'} files={img3Examples.basic} />

### IPFS protocol

`ipfs://` protocol will be automatically replaced with the fastest gateway.

<Playground title={'IPFS protocol - @lxdao/img3'} files={img3Examples.ipfs} />

### Overwrite gateways

You can overwrite the default gateways, will be used in the request, and the fastest gateway will be cached.

<Playground title={'Overwrite gateways - @lxdao/img3'} files={img3Examples.gateways} />
