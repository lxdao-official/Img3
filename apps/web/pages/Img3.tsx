import { Img3 } from 'img3/src';
import { useState } from 'react';

export default function Web() {
  const [show, setShow] = useState(false);
  return (
    <div>
      <h1>Web</h1>
      <div>
        <button
          onClick={() => {
            setShow(!show);
          }}
        >
          Next Img3 by IPFS cache
        </button>
      </div>
      <Img3
        style={{ height: 200, width: 200, border: '1px solid #99a00a' }}
        src={'ipfs://bafkreid67qrfaq2yqacnsvpvfnetjocgy7kiuwu4jw4v23tc3yqgfgis2e'}
      />
      <Img3
        style={{ height: 200, width: 200, border: '1px solid #99a00a' }}
        src={'ipfs://bafkreidpunfemg2foalobiurfob4v6rdb4i3fmujvj5lpnzahfo3a4mxmy'}
        gateway={'https://ipfs.io/ipfs/'}
        size={80}
        color={'#99a00a'}
        timeout={10000}
      />
      {show && (
        <Img3
          style={{ height: 200, width: 200, border: '1px solid #99a00a' }}
          src={'ipfs://bafkreidpunfemg2foalobiurfob4v6rdb4i3fmujvj5lpnzahfo3a4mxmy'}
          color={'#129af9'}
        />
      )}
    </div>
  );
}
