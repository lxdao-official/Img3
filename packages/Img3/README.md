# `Img3`

> TODO: description

## Usage

```
const img3 = require('Img3');

// TODO: DEMONSTRATE API
```
