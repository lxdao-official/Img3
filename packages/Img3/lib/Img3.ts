function img3() {
  return 'Hello from img3';
}

export default img3;
