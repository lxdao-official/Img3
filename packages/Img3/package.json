{
  "name": "img3",
  "version": "0.0.0",
  "description": "> TODO: description",
  "author": "wwei-github <418047394@qq.com>",
  "homepage": "https://github.com/lxdao-official/Img3#readme",
  "license": "ISC",
  "main": "lib/Img3.js",
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/lxdao-official/Img3.git"
  },
  "scripts": {
    "test": "node ./__tests__/Img3.test.js"
  },
  "bugs": {
    "url": "https://github.com/lxdao-official/Img3/issues"
  }
}
