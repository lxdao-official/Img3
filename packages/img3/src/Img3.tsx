import React, { useState, useEffect } from 'react';

import styled, { keyframes } from 'styled-components';

const rotate = keyframes`
  0% { transform: rotate(0deg); }
  100% {  transform: rotate(360deg); }
`;

const Div = styled.div`
  display: inline-block;
  background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAMgAAADICAYAAACtWK6eAAAAAXNSR0IArs4c6QAAFOlJREFUeF7tnWesFcUbxgfF3rtYQBGxK4ooAoIKmCCKUYkFjcEQjVFj1NhioonGWGP8YDAxiugHbJQEiRixiyBYELuCioq9967881sz/C/3nt2dmbN7z92d503upzuzu/O885yZedt0W7FixQojEQJCoCEC3UQQzQwhkI6ACKLZIQQyEBBBND2EgAiiOSAEwhDQChKGm3pFgoAIEomiNcwwBESQMNzUKxIERJBIFK1hhiEggoThpl6RICCCRKJoDTMMAREkDDf1igQBESQSRWuYYQiIIGG4qVckCIggkShawwxDQAQJw029IkFABIlE0RpmGAIiSBhu6hUJAiJIJIrWMMMQEEHCcFOvSBAQQSJRtIYZhoAIEoabekWCgAgSiaI1zDAERJAw3NQrEgREkEgUrWGGISCChOGmXpEgIIJEomgNMwwBESQMN/WKBAERJBJFa5hhCIggYbipVyQIiCCRKFrDDENABAnDTb0iQUAEiUTRGmYYAiJIGG7qFQkCIkgkitYwwxAQQcJwU69IEBBBIlG0hhmGgAgShpt6RYKACBKJojXMMAREkDDc1CsSBESQSBStYYYhIIKE4aZekSAggkSiaA0zDAERJAw39YoEAREkEkVrmGEIiCBhuKlXJAiIIJEoWsMMQ0AECcOtsF4rVqwwf/75p/n777/Nv//+mzx3tdVWM927dzdrrrmm6datW2Hv0oP8ERBB/DErrAeE+P33381ff/3V8JlrrLGGWXvttRPCSFqDgAjSGtyTt/7222/J6pElrCLrrLNOC78y7leLIC3SP6vGr7/+6vT2dddd17CaSDofARGk8zF3Xj3sp2kVaZGSjDEiSIuw//HHHw0HdBfhDLLBBhu4NFWbghEQQRwA/eijj8ybb75pPvjgg+RQvfnmm5s+ffqYAw88MMjKBDEgiI9stNFGPs1XaWvPOv/880/yvWzX1lprLW3bHBAVQXJAWrBggZk+fbp59dVXO7QcPny4GTNmjNl1110doF61yQ8//ODVJ4Qgf/zxh/n555/NL7/80uFdEGTDDTeUASBHCyJIBkALFy40t99+u1m+fHlqq/79+5tx48aZPffc02vCl00QyME7WPHShJVk4403FkkyNCeCZIBz/fXXmyeeeCJ34g8cONCcf/75yWRzlTIJwlbq22+/TczIebLeeuuZzTbbLK9ZtP8XQVJUv2zZMnPppZea77//3mlynHXWWeboo492akujMgny008/me+++87pW1ZffXWz5ZZb6jySgpYIkgLMSy+9ZC677DKnSUajvn37mosuusj07NnTqU9ZBMG/8s033+Q6INt+JATBYy/piIAIkjIr5s+fb6688kqvOXPiiSea0047zalPWQRhxfO1kG2xxRY6h2gFcZq3KxstXrzYXHLJJV6dMP+yivTr1y+3XxkE4UDO6sEZxEe0gqSjpRUkBRtMo+ecc4759NNPfeaaGTlypLnwwgtz+5RBEMjRyKSb9TFEDW+99dYKiNQKkjtnOzSYNGmSeeCBB7w64vVmFTnssMMy+xVNEIgBQXwFX4iP9c33+VVvrxUkQ4Pvv/++ueGGGwwWLR/BN3LxxRdnTrwiCcKWCnJk+TwafT9+ELaFCoTUFstnfq/Sdtq0aYmz0FdOP/10M3bs2NRuRRKEQ7mrObrtB7FysIJIRJDgOcDEu/HGG82LL77o9Ywdd9wx2WrttNNODfsVRRDySVg90pKu0j4asy4OQvwgEhGkqTmANx2S2JRY14cdd9xx5owzzujQvMhgRRyCOAZ9BXLgRZdkI6AziOMMuemmm8ycOXMcW//XjABDVpEBAwZ06FdEuDuhJKwevsRVeIm7GkUQR6zwi7CKfP311449/ms2YsSIhCTtxSXd1vZJS5gKMeuypWL1kOfcTY0iiBtOSavJkyeb++67z6OHSaxEBD1ut912q/RrNuWW/l9++aW3U1BmXS/1xZFRSMITh21CwCmA0Lt3b0Oet68Q9o7Zd8mSJV5dIUgj77rLKpK2emDShSA+wrNYPULMupiS+V62czbpKoZVqNYryKJFiwwxVSQ7tbUa4TkmG3Dw4MGmV69ePnPMzJw509x6661efS6//HIzZMiQDn2aKftDwQff7d4mm2zinboLETEEcGbiB8YKHnjSgDln1dlUXFuCEI07ZcoU88knn6ROZsgxdOjQ5M9VyXisWUXINHSVa665xuA8bCShheN8VxBWTlYP1xpbFLKDGPxlOSDJTOzRo4czfq6YdZV2tSQIpLjjjjuSPHIXYfsDSVhRXGTu3Lnm6quvdmlqttpqK3PLLbckv7RFCqvP559/nlRkdBHOQq7bSrajPuZjrGKcsSBL3aSWBMFvcdttt3npil9Wu5rsscceuX15/owZM3LbHX/88WbChAm57UIauIa2sxVie5Un5K9DDJ7rWnHFPhOCbLrppnmvqNz/a0mQu+++28yePTtIGWxDLFG22Wab1GewJ8eqlfUetlVnnnmmcxKV7wdjycrbAq2//vpJTFjW1oqzhd1O+Xrk7TezQmXh5Tu2rtK+lgTh190llzxLCZT1OfjggxOypG1NOCQ//fTTZt68eeaNN95Y+Ti2HIS9jxo1yuywww6l6poJbSuXtHUYst3hu/lLCyfBMmVXDNcqj2mDYfVob8oudeCd9PBaEoQAw6lTpxYC4f7775+QBKtXmjDJOPdgdsWECrk4uHamcBYhLoutERYm/rLirLDqWetUEd/JWYu/ukktCUJgIV7vIuXII480xxxzjGHLUmVh1YDIX331VaHDYKV0tQQW+uKSH1ZLgvArOnHiRIO1qUhh23TsscdW9jBqve8hiVVZOGIA2H777YuEuss8q5YEAd2lS5cm2YCNKiI2g/6pp55qRo8e3cwjWtaXM5NvCnHex2IhY2vlakLOe15X+39tCQLQb731lpk1a5bBaViUUGb0ggsuKNyvUdT3pT2HM8qHH37onbOe9V1sqaiIUuew+VoTBOVyGCXc5JlnnjGk0BYhV111ldlll12KeFSnPYMIgPfee6+Q9+GVZ1uF+RhjQJ2l9gSxyvv4448TkvDnWnUwTfHnnnuus9e9q0wenH8EbTYjWOggBn919Jo3wiYagtjBv/766yuJ4ustts9gi5Vl9m1mEpbVl5WULVaoWGJU3YrnO/7oCGIBwsLFauJ7iGe/fe2111bO5o+PBMOFb1E5G6YSa2mgaAkCUdh2QBLI4rr9wDs+fvx43x+iLtEeC5ZriDy5HnbVqPs5I0s5URPEAsPNUfZ8klUAYZ999jGnnHJKabFVZbOIsPXPPvsss8gDZGC1gBy6XVd3FK4yJ19++eXE4sU5hfs1rBCIN2jQIDNs2LDKxxvZBChWz7aBiRzAbWCj7kP8/7TQCtLgZ5u4KixdmEbZalDjqm5hFPhFbAotMVv23sKyV7GqPV8EqZrG9L2dioAI0qlw62VVQ0AEqZrG9L2dioAI0qlw62VVQ6AWBCGT7pVXXkn+yOzzLQxtlUZhBa5z3nvvvQ0mXdcKIFVTevvvJaKAEHgsd/zhVAwR6m6RWcgfqcvUz6q6VJ4gVPYgvfapp54qVBeHHHJIcgkONbTqLKTa4kAsOgye/HT+qh4GX2mCkItNKdDnn3++lDl8wAEHGC7mrGv8EX4QInx9KzS6gs3dh1z/EFLJ0fUdZberNEEef/xxc//995eK0QknnGCGDx9e6jta9XD8PUWFwKeNAYJsu+22rRpi0++tNEFuvvnmJCmqTNl9993NeeedV+YrWvZsAjVDbqby+WBCVvbaay+fLl2qbWUJQi2nK664ouncjjxtcOC87rrr8ppV7v9E9VLcom293TIGQd5I1VID2uIgguTMChGkOdqIIM3h11RvbbGagi/JhdEWKxvDyq4gDEuH9OYIokN6Pn6VJojMvPkKzmqBmffdd98tvIicfafMvM3pp5DeJAA9+eSTchQGoilHYY23WHZoRYaacPUBYSYxhpoQbkIeTDOhJph1CTNRqEngL5a6CYEqIVDpM0iVgNa3VhMBEaSaetNXdxICIkgnAa3XVBMBEaSB3rg7g5wSihrgCSbYrm4FmjmIM0aKNxBtS3GKukYtN0NNEaQNem+//bZZvHhx4htom3SFZQarFrdNVf0WJep+UacYJyE/AFYgCOWN+DHAfyH5DwERxJgkWYgrEvij1E+aUNGdm6aqmkTFxaNcjZ11uxRZgdw1yIU4dSt1FEL6qAnCr+miRYsSYuBwdBEu9hwzZoxL0y7XhoJ4y5Ytc/ouyAFR6nr/uRMIMa8glhhLlixxxSppRzlO8kOqdic4HnPKq/pe88wFOZYoXkDVpHF0KwjnC7udCr3+gGvYqpYExApJ/keIUHzBkoRzSkwSDUHIu7bECK16YifGuHHjzL777lupecKhnFWzGeEgb88nsVi8ak8QIn6xTEEOrDdFyNlnn2249rhKQjmfefPmFfLJVH+HKFQtqftNU7UmCAdSygFhuSlKKGTNFqtqv6Ck1rLFalu1vllMMHn36dOncucxn3HXliBciPPII48Y30N4HnhHHXWUGTp0aF6zLvl/LjGlsF6RwiEe8ze+ojpKLQnC4Zt6Wc3uudsrfODAgWbEiBGVuwLajoO7QfjBaOauwkYkYLtVtTOZK5lrSRC2VJMnT3bFwKkdqwb1sapeKdDeVVjUldgWPIrsVT3KoNFEqCVBHn30UTNnzhyniZ/XiASq/v37Z5p18b6zt8dTzWU0eNo7+9JLjBF8A8ljxI1xSxRWpzTB7Lt8+XLzxRdf5EHg9H+2WX379nVqW6VGtSTItGnTzMKFC5vSA6EWEIO/tImGJ57V6p133kkmmxXaUwC7X79+hj16mUJVEnw7rAhtMwGJp2IMvXr1Sl31cBpi2eOv2eomvIcx101qSZCZM2eaZ599NkhXVHi3xMgK2iPQDwtZ1jmnd+/eZuTIkUkQYBmCPwfzdVbh6Z133jkhapY5ltUHkkByzikhwlhZbesmtSQIxaynTp3qpSuuOoAY++23X2K6zJPHHnvMaZU66KCDkirxZQjkdDFh77bbbsnY8oQrou2K4htlQLRzz549815Ruf/XkiDsq2fMmJFsO1yE/TMTyNUSQ1j89OnTXR6dWLwmTJjgdKWyjZNyqYaOX2P27NmZ0cdtPxAjg+sExuvOapIV9dv22RRoYHtVNd+QiwJrSRAGzi/rQw89lFnav0ePHiu3U67KZWI++OCDXv6Vk046ybAFaSRsaTjDcFe7PQfgU2BPD3HTzj9sq7gXxVUwxQ4ePNj5KgLGaVcTjA9pAm4U+K6jBYsx15YgDI7K79x9vnTpUsM+2wpnAm6SYsUgXMJH8EbjgPSRsWPHJpO9vVBi54UXXki+r5FgFSJJq5ETDkco0bk+MmDAgIbfkfUMyMGKgtWrba4MZxoMEOBXV3LUniBW8SgXixNWHhtwl2UCTZsw1I1i9fC9jenkk0/uELuFOXbu3LmGHI0sgchsj9pfZ8bNWpyDfIStEKtISCIU2z+boospG4KEPMfne7tC21qvIEUDjNXKN+APfwQEYXK2Fc5HDz/8sNMnjho1qsMWjV92CEKeh49AOKxaEjcERBA3nJLwDMzHrEQ+Qt5IowxECm9z2HcRrFCNLGHz5893NkTY9xAJwCpS522RC6aubUQQR6RmzZqVXBfgI0xGyME1ZG0FE+pdd93l/OuPZ3z8+PEdXs3ZAJL4XoLD92B+luQjIILkY5ScEzh7+PoGuFmJ4Mb2gpPxzjvvdHjz/5uQg9JIcBT6XkPHeYZVpGo5LV6AFdRYBMkBEssN5HD1qdjH4YVn9UjbykycONFLhWkEIf6KVcQ3VATr06BBgzLjtbw+sKaNRZAcxS5YsCC5qMdXiPwlPD5NiiIIz8fnExLaj3OU840kHQERJGN24JFn9fC9RxynIKtHVjXGIgnClo1VxLV0kR0y/hVWkbomOxVBfBEkA0U81c8995wXzuzvIQfm1CwpkiC8h/RiXxM0/QgwdA2x8QKiJo1FkBRFYhmaNGmS9zXTxCSRlpsnRROE94WYffHTHHHEEc4hKHnjqtv/RZAUjRIbNWXKFC99M9lYPVysQ2UQBO86q0jbmrsuA8DSVtVyqi7ja6aNCJKCHgGEJF75CPv5Qw891KlLGQThxcSe+RZmGDZsWJJcJemIgAiSMisw6957773Oc4bIYFYP1+SosghCvBSriE95HyxufL9EBHGeA1iu7rnnHud8i8MPP9wQLesqZRGE9xPC4lpmlKBNtlidnUPvilOr22kFydAAsVd50bZ0J6119OjRXpfslEkQckzw37hUkqQQHl51SWMERJCMmUGeBk5CwtzTBJ/HkCFDvPfwZRKEb2UFfO211zJ9I2Q7kmLMpTkSESRoDkASfo1JUGov+DpIaAqZYGUThG8lxxxjQ6M7QQiBwYtOpqEkHQGtIA6zwxYzIEebpCsShZhgbK3aJzI5PC4xwxYVrJj3PgIsifrl0E7uSPfu3ZPzBtmArCCSbAREkBbMkKLC3Vvw6dG9UgRpkcqLSJhq0adH9VoRpEXqbjbltkWfHd1rRZAWqbzZog0t+uzoXiuCtFDleWV/MALgfFQ4euuUJIK0DvvkzaGF41r82dG8XgSJRtUaaAgCIkgIauoTDQIiSDSq1kBDEBBBQlBTn2gQEEGiUbUGGoKACBKCmvpEg4AIEo2qNdAQBESQENTUJxoERJBoVK2BhiAggoSgpj7RICCCRKNqDTQEAREkBDX1iQYBESQaVWugIQiIICGoqU80CIgg0ahaAw1BQAQJQU19okFABIlG1RpoCAIiSAhq6hMNAiJINKrWQEMQEEFCUFOfaBAQQaJRtQYagoAIEoKa+kSDgAgSjao10BAERJAQ1NQnGgREkGhUrYGGICCChKCmPtEgIIJEo2oNNAQBESQENfWJBgERJBpVa6AhCIggIaipTzQIiCDRqFoDDUFABAlBTX2iQUAEiUbVGmgIAiJICGrqEw0CIkg0qtZAQxAQQUJQU59oEBBBolG1BhqCgAgSgpr6RIOACBKNqjXQEAREkBDU1CcaBESQaFStgYYgIIKEoKY+0SAggkSjag00BAERJAQ19YkGAREkGlVroCEIiCAhqKlPNAiIINGoWgMNQeB/nYkQTsN6jlcAAAAASUVORK5CYII);
  background-size: 60px 60px;
  background-position: center;
  background-repeat: no-repeat;
  animation: ${rotate} 2s linear infinite;
`;

export const Img3 = (props: {
  style?: React.CSSProperties;
  alt?: string;
  gateway?: string;
  className?: string;
  src: string;
}) => {
  const { style } = props;

  const [loadingState, setLoadingState] = useState('loading');

  const styled = { display: 'inline-block', ...style };

  return <Div style={styled}>1231</Div>;
};
