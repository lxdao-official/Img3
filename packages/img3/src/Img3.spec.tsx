/**
 * @jest-environment jsdom
 */

import React from "react";
import { render } from "@testing-library/react";
import { Img3 } from "./Img3";

test("renders the Img3", () => {
  render(
    <Img3 src="ipfs://bafkreid67qrfaq2yqacnsvpvfnetjocgy7kiuwu4jw4v23tc3yqgfgis2e" />
  );
});
