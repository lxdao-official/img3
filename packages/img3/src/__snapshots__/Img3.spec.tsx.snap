// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Img3 should render error when not found 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe bHDyYL"
  >
    <span />
  </div>
</DocumentFragment>
`;

exports[`render Img3 should render error when timeout 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe bHDyYL"
  >
    <span />
  </div>
</DocumentFragment>
`;
